core/src/main/scala/org/bitcoins/core/api/Callback.scala

```diff
@@ -0,0 +1,56 @@
+package org.bitcoins.core.api
+
+import org.bitcoins.core.util.{FutureUtil, SeqWrapper}
+import org.slf4j.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+/** A function to be called in response to an event */
+trait Callback[T] {
+  def apply(param: T): Future[Unit]
+}
+
+/** A function with two parameters to be called in response to an event */
+trait CallbackBinary[T1, T2] extends Callback[(T1, T2)] {
+  def apply(param1: T1, param2: T2): Future[Unit]
+
+  override def apply(param: (T1, T2)): Future[Unit] = apply(param._1, param._2)
+}
+
+/** A function with three parameters to be called in response to an event */
+trait CallbackTernary[T1, T2, T3] extends Callback[(T1, T2, T3)] {
```

Comment:
Why wouldn't we name it `Callback3` similar to how it's done in the standard library: https://www.scala-lang.org/api/current/scala/Function3.html ?